### PR TITLE
[feat] Nullable fields

### PR DESCRIPTION
**Pull request by erfanium**
Related #113

### CODE REVIEW

File: test/integration.spec.js [hunk 1]
```diff
@@ -1032,3 +1032,86 @@ describe("Test default value sanitizer", () => {
 	});
 });
 
+describe("Test optional option", () => {
```
Comment (icebob) [inline test/integration.spec.js:1035]: Why these both tests are duplicated?

File: test/integration.spec.js [hunk 2]
```diff
@@ -1032,3 +1032,86 @@ describe("Test default value sanitizer", () => {
...
+	it("should not throw error if value is null", () => {
+		const o = { foo: null };
+		expect(check(o)).toBe(true);
+		expect(o.foo).toBe(null);
+	});
+
+	it("should not throw error if value exist", () => {
+		expect(check({ foo: 2 })).toBe(true);
+	});
+});
```
Comment (icebob) [inline test/integration.spec.js:1117]: Please add test when 
- `nullable: true` and there is `default`
- `optional: true` and there is `default`
- `nullable: true` and `optional: true`
- `nullable: true` and `optional: true` and there is `default`

**Review by icebob — CHANGES_REQUESTED** [review]

File: lib/validator.js [hunk 3]
```diff
@@ -97,41 +97,38 @@ class Validator {
...
+		let handleNoValueSrc;
+
+		let skipUndefinedValue = rule.schema.optional === true || rule.schema.type === "forbidden";
+		let skipNullValue = rule.schema.optional === true || rule.schema.nullable === true || rule.schema.type === "forbidden";
+
+		if (rule.schema.default != null) {
+			// We should set default-value when value is undefined or null, not skip!
+			skipUndefinedValue = false;
+			skipNullValue = false;
+			let defaultValue = JSON.stringify(rule.schema.default);
```
[the remark this answers is not in the document]
Comment (erfanium) [reply lib/validator.js:109]: Thank!

File: test/integration.spec.js (hunk 1 above)
Comment (erfanium) [reply test/integration.spec.js:1035]: I have no answer😁😁

**Review by icebob — APPROVED** [review]
Thanks!

**Comment by icebob** [comment]
@erfanium could you ping me on Discord chat, please? https://discord.gg/TSEcDRP

**Pull request merged**